src/connector.js

```diff
@@ -901,16 +912,12 @@ function updateQuerySummaryState( newState ) {
 		querySummaryElement.textContent = "";
 		if ( querySummaryState.total > 0 ) {
 			let numberOfResults = querySummaryState.total.toLocaleString( params.lang );
-			// Create the <h2> element
-			const hTwoAnchor = document.createElement("h2");
 			// Generate the text content
-			const querySummaryText = ((querySummaryState.query !== "" && !params.isAdvancedSearch) ? querySummaryTemplateHTML : noQuerySummaryTemplateHTML)
-				.replace('%[numberOfResults]', numberOfResults)
-				.replace('%[query]', querySummaryState.query)
-				.replace('%[queryDurationInSeconds]', querySummaryState.durationInSeconds.toLocaleString(params.lang));
-			hTwoAnchor.textContent = querySummaryText;
-			querySummaryElement.innerHTML = "";
-			querySummaryElement.appendChild(hTwoAnchor);
+			const querySummaryHTML = ( ( querySummaryState.query !== "" && !params.isAdvancedSearch ) ? querySummaryTemplateHTML : noQuerySummaryTemplateHTML )
+				.replace( '%[numberOfResults]', numberOfResults )
+				.replace( '%[query]', querySummaryState.query )
```

Comment:
One suggestion: This line could replace `%[query]` by an empty `span` with a class like `sr-query`, such as: `<span class="sr-query"></span>`, and after the `querySummaryHTML` is inserted to the `querySummaryElement` element with  `innerHTML`, then appended to the DOM, then the new `span` element with the class can be targeted and the value of `querySummaryState.query` can be inserted to it with `textContent`.